client/src/DbtLanguageClientManager.ts

```diff
@@ -25,13 +25,15 @@ export class DbtLanguageClientManager {
     previewContentProvider.onDidChange(() => this.applyPreviewDiagnostics());
   }
 
-  async applyPreviewDiagnostics(): Promise<void> {
+  applyPreviewDiagnostics(): void {
     const previewDiagnostics = this.previewContentProvider.getPreviewDiagnostics();
-    const activeClient = await this.getClientForActiveDocument();
 
     for (const client of this.clients.values()) {
       const clientDiagnostics = client.getDiagnostics();
-      clientDiagnostics?.set(SqlPreviewContentProvider.URI, client.getProjectUri() === activeClient?.getProjectUri() ? previewDiagnostics : []);
+      clientDiagnostics?.set(
+        SqlPreviewContentProvider.URI,
+        this.previewContentProvider.activeDocUri.fsPath.startsWith(client.getProjectUri().fsPath) ? previewDiagnostics : [],
```

Comment:
If we can create some e2e test for this fix?

Reply:
I thought about it, It could be problematic because we need to click on elements out of the document and we have no such API.